### PR TITLE
[Addon] add application template, template->resources

### PR DESCRIPTION
**Pull request by chivalryq**
### Checklist

I have:
- [ ] Title of the PR starts with OAM type (e.g. `[Workload]` or `[Trait]` or `[Scope]`).
- [ ] Updated/Added any relevant [documentation] and [examples].
- [ ] Unit/E2E Tests added.

### CODE REVIEW

File: addons/example/readme.md [hunk 1]
```diff
@@ -4,9 +4,10 @@ This is an example addon based [FluxCD](https://fluxcd.io/)
 
 ## Directory Structure
 
+- template.yaml: contains the basic app, you can add some component and workflow to meet your requirements
 - metadata.yaml: contains addon metadata information.
-- definitions/: contains the X-Definition yaml/cue files.
-- template/:
+- definitions/: contains the X-Definition yaml/cue files. These file will be rendered as KubeVela Component in `template.yaml`
+- resources/:
```
Comment (hongchaodeng) [inline addons/example/readme.md:10]: 直接叫 components 好了

File: addons/example/readme.md [hunk 2]
```diff
@@ -4,9 +4,10 @@ This is an example addon based [FluxCD](https://fluxcd.io/)
 
 ## Directory Structure
 
+- template.yaml: contains the basic app, you can add some component and workflow to meet your requirements
```
Comment (hongchaodeng) [inline addons/example/readme.md:7]: 还需要说清楚其他那些渲染的 components 怎么加进来的。

File: addons/example/readme.md (hunk 1 above)
Comment (chivalryq) [reply addons/example/readme.md:10]: 可是definitions渲染出来也是components

File: addons/example/readme.md (hunk 2 above)
Comment (chivalryq) [reply addons/example/readme.md:7]: right

File: addons/example/readme.md (hunk 1 above)
Comment (hongchaodeng) [reply addons/example/readme.md:10]: 有道理。。

Comment (chivalryq) [reply addons/example/readme.md:10]: 🤪

**Review by hongchaodeng — APPROVED** [review]

**Pull request merged**